### ISSUE TITLE
Hybrid solver not working for Opiod_Model

### ISSUE THREAD

**Issue by hodgespodge** [GillesPy2/GillesPy2#320]
When Opiod_Model.ipynb is run using the Basic Tau Hybrid Solver, you get a type error in __f() 

Tested on a windows machine: 

```
Exception in thread Thread-6:
Traceback (most recent call last):
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Samuel Hodges\PycharmProjects\GillesPy2\gillespy2\solvers\numpy\basic_tau_hybrid_solver.py", line 974, in __run
    event_sensitivity, tau_step, pure_ode, debug)
  File "C:\Users\Samuel Hodges\PycharmProjects\GillesPy2\gillespy2\solvers\numpy\basic_tau_hybrid_solver.py", line 567, in __simulate
    pure_ode)
  File "C:\Users\Samuel Hodges\PycharmProjects\GillesPy2\gillespy2\solvers\numpy\basic_tau_hybrid_solver.py", line 460, in __integrate
    events=tau_event, **integrator_options)
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\site-packages\scipy\integrate\_ivp\ivp.py", line 577, in solve_ivp
    message = solver.step()
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\site-packages\scipy\integrate\_ivp\base.py", line 182, in step
    success, message = self._step_impl()
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\site-packages\scipy\integrate\_ivp\lsoda.py", line 150, in _step_impl
    self.t_bound, solver.f_params, solver.jac_params)
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\site-packages\scipy\integrate\_ode.py", line 1343, in run
    y1, t, istate = self.runner(*args)
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\site-packages\scipy\integrate\_ivp\base.py", line 139, in fun
    return self.fun_single(t, y)
  File "C:\Users\Samuel Hodges\AppData\Local\Programs\Python\Python37\lib\site-packages\scipy\integrate\_ivp\base.py", line 21, in fun_wrapped
    return np.asarray(fun(t, y), dtype=dtype)
  File "C:\Users\Samuel Hodges\PycharmProjects\GillesPy2\gillespy2\solvers\numpy\basic_tau_hybrid_solver.py", line 443, in <lambda>
    rhs = lambda t, y: BasicTauHybridSolver.__f(t, y, *int_args)
  File "C:\Users\Samuel Hodges\PycharmProjects\GillesPy2\gillespy2\solvers\numpy\basic_tau_hybrid_solver.py", line 219, in __f
    state_change[y_map[rr]] += eval(compiled_rate_rules[rr], {**curr_state, **eval_globals})
  File "<string>", line 1, in <module>
TypeError: unsupported operand type(s) for *: 'builtin_function_or_method' and 'float'
```

**Comment by seanebum** [comment]
This is caused by using variable named "gamma" which overlaps with the evaluation definitions from math.__dict__ for math.gamma.   I can think of a few different ways to prevent this from happening... we could:

1. Restrict use of variable names that overlap with math.__dict__
2. Override math definitions if the name is used as a variable
3. Check for overlaps when adding items, and rename them by adding something like an '_'

**Comment by seanebum** [comment]
Fixed with #322
